gdx-backend-jtransc/src/com/jtransc/media/limelibgdx/LimeInput.java

```diff
@@ -443,17 +449,9 @@ private static int convertKeyCode(int i) {
 
 	private boolean checkKeyArray(boolean[] array, int i) {
 		if (i < 0) {
```

Comment:
What about (can't fail either):

```
final int actualIndex = (i < 0) ? 0 : i;
return actualIndex < array.length && array[actualIndex];
```
Just one return, array access just once. Probably easier to understand. In the if you are still checking for bounds (with reversed condition) and inlining the index that you now want to use 0.

Or (more restrictive):

```
if (i < 0) throw new RuntimeException("Invalid index");
return i < array.length && array[i];
```